validation/src/header.rs

```diff
@@ -86,10 +87,44 @@ pub fn validate_header(
     Ok(())
 }
 
+#[cfg(target_arch = "wasm32")]
+fn get_current_time_in_secs() -> u64 {
+    // to get seconds from nanoseconds
+    ic_cdk::api::time() / 1_000_000_000
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+fn get_current_time_in_secs() -> u64 {
+    use std::time::SystemTime;
+
+    SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .unwrap()
+        .as_secs()
+}
+
+fn block_timestamp_less_than_2h_from_current_time(block_time_secs: u64) -> bool {
+    let current_time_secs = get_current_time_in_secs();
+    if block_time_secs <= current_time_secs {
+        return true;
+    }
+    let time_diff_secs = block_time_secs - current_time_secs;
+    let number_of_seconds_in_one_hour = 60 * 60;
+    let time_diff_hours = time_diff_secs / number_of_seconds_in_one_hour;
+    time_diff_hours < 2
+}
+
 /// Validates if a header's timestamp is valid.
 /// Bitcoin Protocol Rules wiki https://en.bitcoin.it/wiki/Protocol_rules says,
 /// "Reject if timestamp is the median time of the last 11 blocks or before"
-fn is_timestamp_valid(store: &impl HeaderStore, header: &BlockHeader) -> bool {
+/// "Reject if the timestamp of the block is < 2 hours from the current time"
```

Comment:
I'd quote what the page above states exactly: "Block timestamp must not be more than two hours in the future".